### PR TITLE
wc_sessionPropose method documentation

### PR DESCRIPTION
**Pull request by dukevantreta**
Adds reference documentation for the `wc_sessionPropose` request and response JSON payloads.

Check if all parameters are correct, if there are any optional fields missing or not labelled.

Add comments if there are still any unresolved questions.

### CODE REVIEW

File: wc-rpc-methods.md [hunk 1]
```diff
@@ -14,10 +15,60 @@ error: {
...
+        }
+    }
+}
+```
+
+```jsonc
+// Success result
+{
+    "relay": {
+        "protocol": string,
```
Comment (dukevantreta) [inline wc-rpc-methods.md:64]: Why does the responder send a `relay` field? What happens if the value is not the same as the proposal?

File: wc-rpc-methods.md [hunk 2]
```diff
@@ -14,10 +15,60 @@ error: {
...
+            "icons": [string]
+        }
+    }
+}
+```
+
+```jsonc
+// Success result
+{
+    "relay": {
```
Comment (dukevantreta) [inline wc-rpc-methods.md:63]: Is the response's `relay` field optional or required?

File: wc-rpc-methods.md [hunk 3]
```diff
@@ -14,10 +15,60 @@ error: {
...
+            "types": [string]
+        },
+        "controller": {
+            "publicKey": string
+        }
+    },
+    "proposer": {
+        "publicKey": string,
+        "controller": bool,
+        "metadata": {
```
Comment (dukevantreta) [inline wc-rpc-methods.md:50]: Are all `metadata` fields required? Is any of them optional?

File: wc-rpc-methods.md [hunk 4]
```diff
@@ -14,10 +15,60 @@ error: {
...
+    "permissions": {
+        "jsonrpc": {
+            "methods": [string]
+        },
+        "notifications": {
+            "types": [string]
+        },
+        "controller": {
+            "publicKey": string
+        }
```
Comment (jakubuid) [inline wc-rpc-methods.md:45]: Should controller be nested in permissions?

Comment (pedrouid) [reply wc-rpc-methods.md:45]: Agreed, controller should be removed here

File: wc-rpc-methods.md (hunk 3 above)
Comment (pedrouid) [reply wc-rpc-methods.md:50]: Yes, all 4 fields are required

File: wc-rpc-methods.md (hunk 2 above)
Comment (pedrouid) [reply wc-rpc-methods.md:63]: Required

File: wc-rpc-methods.md (hunk 1 above)
Comment (pedrouid) [reply wc-rpc-methods.md:64]: Proposer would use the new relay instead

We should make the relay an array on proposal so that responder can choose

Comment (dukevantreta) [reply wc-rpc-methods.md:64]: Changed the payload to use an array and added a bullet point to describe the selection.

Can the responder send a relay value totally different from the options proposed?

File: wc-rpc-methods.md (hunk 4 above)
Comment (dukevantreta) [reply wc-rpc-methods.md:45]: Removed.

**Pull request merged**